### PR TITLE
Add new license detection HESSLA

### LINKED ISSUE #2210: Validate that all licenses (and their URLs) are detected from https://directory.fsf.org/wiki?title=Category:License
There are about ~240 URLs and licenses listed at https://directory.fsf.org/wiki?title=Category:License
We should:
1. ensure that we can detect all of them
2. ensure that we can detect their URL
We should also add as an "other_urls" ech URL to the corresponding license YAML

[fsf-license-urls.txt](https://github.com/nexB/scancode-toolkit/files/5177958/fsf-license-urls.txt)

### PR DESCRIPTION
**Pull request by sankha555**
Fixes #2210

### Tasks

* [x] Reviewed [contribution guidelines](https://github.com/nexB/scancode-toolkit/blob/develop/CONTRIBUTING.rst)
* [x] PR is descriptively titled 📑 and links the original issue above 🔗
* [x] Tests pass -- look for a green checkbox ✔️ a few minutes after opening your PR
  Run [tests](https://scancode-toolkit.readthedocs.io/en/latest/contribute/contrib_dev.html#running-tests) locally to check for errors. 
* [x] Commits are in uniquely-named feature branch and has no merge conflicts 📁

This PR adds new license text hessla.LICENSE  and corresponding YAML file to the dataset of licenses. This will enable detection of the HESSLA license which has been mentioned in the wiki page mentioned in the issue statement. The key used for the license is hessla and I have categorised it into proprietary licenses. Based on the reviews received for this PR, I will continue work to incorporate the remaining licenses mentioned in the wiki.

- Sankha Das

### CODE REVIEW

File: src/licensedcode/data/licenses/hessla.yml [hunk 1]
```diff
@@ -0,0 +1,8 @@
+key: hessla
+short_name: HESSLA 
+name: Hacktivismo Enhanced-Source Software License Agreement
+category: Proprietary
```
Comment (pombredanne) [inline src/licensedcode/data/licenses/hessla.yml:4]: Use instead:
```suggestion
category: Proprietary Free
```

File: src/licensedcode/data/licenses/hessla.yml [hunk 2]
```diff
@@ -0,0 +1,8 @@
...
+owner: Hacktivismo
+homepage_url: http://www.hacktivismo.com/about/hessla.php
+text_urls: https://directory.fsf.org/wiki/License:HESSLA
```
Comment (pombredanne) [inline src/licensedcode/data/licenses/hessla.yml:7]: This should be a list not a string:
```suggestion
text_urls:
    - https://directory.fsf.org/wiki/License:HESSLA
```

**Review by pombredanne — CHANGES_REQUESTED** [review]
Thank you++ for this. See my comments inline for your review.

**Comment by sankha555** [comment]
@pombredanne Just incorporated the changes submitted by you. Please have a look at it. Also, if you think my approach to the issue is correct, then I may proceed with the rest of the licences.

Also, I had a few queries regarding the license score matching to 100. Is there a threshold to which I should allow the license score to drop (eg. is 98.7 same as 100 ?) or should it be strictly 100. In case the license score is too low, and I suspect it to be a variation of an existing license, shall I create a new license with the present one or amend the previously existing license text? These pointers will help me to move forward more confidently with the work (and I have already started to enjoy the task! :) ). Thanks!

**Comment by AyanSinhaMahapatra** [comment]
So @sankha555 a few words on the `score`, in most cases it is calculated as `match_coverage * rule_relevance = score`, where `rule_relevance` is 100 by default, if not provided in the `.yml` file, and `match_coverage = matched_length / rule_length`. So if the `match_coverage` is less than 100, there are more some words in the RULE, that wasn't present in the document you scanned.

Additionally, if `score != match_coverage * rule_relevance`, that means that every word in the RULE was matched, but there were some extra words in the file you scanned, in addition to the RULE, and for that there was a imperfect score. 

This [README](https://github.com/nexB/scancode-toolkit/blob/develop/src/licensedcode/README.rst) file has details on the license matching, if you want more details. Also if you have any confusion in any particular case, upload the file and full scan results of that file here, and we'll take a look.

**Comment by sankha555** [comment]
@AyanSinhaMahapatra thanks a lot for the clarifications. I'll take a note of them and will try to upload more licenses in the next PRs.

**Comment by pombredanne** [comment]
I was waiting for the tests to pass to merge.... but the failure is not yours and has been fixed in develop.
All good and merging now! Thanks!

**Pull request merged**